ortho_seq_code/orthogonal_polynomial.py

```diff
@@ -930,126 +924,22 @@ def orthogonal_polynomial(
     print("Trait values estimated from regressions")
     print(Fest)
 
-    ## Graph of regression
-    # Flatten data
-    rFon1D_flat = list(rFon1D.flatten())
-    if any(i != 0 for i in rFon1D_flat):
-        data_null = np.where(
-            np.array(rFon1D_flat) == float(0), float("nan"), rFon1D_flat
-        )
-
-        # Constants/constant arrays
-        ind = np.arange(sites)  # x-axis
-        num_dm = np.arange(dm)
-        width = 1 / sites
-        s = sites * dm
-
-        # Re-vectorization with null values
-        dim_num = dict()
-        for i in ind:
-            dim_num[i] = [data_null[j] for j in np.arange(dm * i, dm * i + dm)]
-        # some_dim = [data_array_flat[i], i for i in range(0, 160, 4)]
-
-        # Remove all null data
-        dim_na = dict()
-        dim_loc = dict()
-        for i in ind:
-            dim_na[i] = np.array(dim_num[i])[np.array(np.isnan(dim_num[i])) == False]
-            dim_loc[i] = np.arange(len(dim_num[i]))[
-                np.array(np.isnan(dim_num[i])) == False
-            ]
-        # Color dictionary with corresponding letters
-
-        dim_aa = dict()
-
-        for i in num_dm:
-            dim_aa[i] = [data_null[j] for j in range(i, s, dm)]
+    ## Bar plot of regression
 
-        col_len = len(colors)
-        alpb_d = dict()
-        if alphbt_input is None:
-            for i in num_dm:
-                if any(i != 0 and i for i in dim_aa[i]):
-                    alpb_d[i] = colors[i % col_len]
-                    alpb_d[alphabets[i]] = alpb_d.pop(i)
+    if custom_aa is not None:
```

Comment:
For these checks I think you can replace them with:

```
alphbt_input = custom_aa or alphabets
rFon1D_o = rf1d(rFon1D, alphbt_input, molecule=molecule, phenotype=pheno_name)
```
`phenotype` you can just assign since it either has a value or is set to `None` which is the default value anyway so it's the same thing.

Reply:
Makes sense, done